projects/laji-ui/src/lib/modal/modal/modal.component.ts

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable @angular-eslint/no-output-on-prefix */
+import { DOCUMENT } from '@angular/common';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, HostListener, Inject, Input, OnDestroy, Output, Renderer2,
+  ViewChild } from '@angular/core';
+import { filter } from 'rxjs/operators';
+
+export type ModalSize = 'sm' | 'md' | 'lg' | 'xl';
+
+@Component({
+  selector: 'lu-modal',
+  template:`
+    <div class="lu-backdrop"></div>
+    <div class="lu-modal-container" [class]="'lu-modal-' + _size" *ngIf="isShown" #container>
+      <lu-button-round *ngIf="!noClose" (click)="hide()" role="neutral" class="lu-modal-close-button">
+        <lu-icon type="close"></lu-icon>
+      </lu-button-round>
+      <div [class]="['lu-modal-content', contentClass]" role="dialog">
+        <ng-content></ng-content>
+      </div>
+    </div>
+  `,
+  styleUrls: ['./modal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ModalComponent implements OnDestroy {
+
+  protected _size: ModalSize = 'md';
+
+  /**
+   * One of 'sm', 'md', 'lg', 'xl'. Defaults to 'md'
+   */
+  @Input() set size(size: ModalSize) {
+    this._size = size;
+  }
+
+  // null becuase undefined in the template [class] causes error.
+  @Input() contentClass: string | null = null;
+
+  /**
+   * Modal can't be closed by clicking backdrop and the close button is hidden.
+   * The parent component must handle closing the modal manually.
+   */
+  @Input() noClose = false;
+
+  /**
+   * Emits true/false according to when the modal shows/hides.
+   */
+  @Output() onShownChange = new EventEmitter<boolean>();
```

Comment:
The naming is now the same as `onHide` and `onShow`, which are from the `ngx-bootstrap` API. And even if we renamed those, it would collide with the `show()` and `hide()` methods so there's no perfect solution :/